lib/active_model_serializers/adapter/json_api/relationship.rb

```diff
@@ -34,6 +34,11 @@ def as_json
 
         private
 
+        def ensure_valid!(json)
```

Comment:
The schema is the only thing we have against which to actually validate documents. The schema must be fixed before changes like this can be merged.